extension/content/engineUrlView.mjs

```diff
@@ -19,22 +25,23 @@ export default class EngineUrlView extends HTMLElement {
     let urls = await browser.experiments.searchengines.getEngineUrls(config);
     let sortedUrls = ROW_HEADERS.map((key) => urls[key]);
 
-    let fragment = this.createTableFragment(
-      COL_HEADERS,
-      ROW_HEADERS,
-      sortedUrls
-    );
-
-    this.appendChild(fragment);
+    if (!this.#table) {
+      this.createTableFragment(COL_HEADERS, ROW_HEADERS);
+    }
 
-    return fragment;
+    this.updateUrls(`Full URL for ${config.name}`, sortedUrls);
   }
 
-  createTableFragment(colHeaders, rowHeaders, sortedUrls) {
+  createTableFragment(colHeaders, rowHeaders) {
     let fragment = document.createDocumentFragment();
-    let table = document.createElement("table");
-    let thead = table.createTHead();
-    let tbody = table.createTBody();
+    this.#table = document.createElement("table");
+    let colgroup = document.createElement("colgroup");
+    colgroup.appendChild(document.createElement("col"));
+    colgroup.appendChild(document.createElement("col"));
+    this.#table.appendChild(colgroup);
```

Comment:
The colgroup is not doing anything functional in the code. From what I can tell, perhaps they were created to have a label for each column. But it seems unnecessary to have these unless they're being used somewhere else. 

I think it's safe to remove the colgroup and associated <col> elements from the code to simplify it without impacting the table's functionality or appearance.

Reply:
I think I might have originally had it for css purporses, but looks like it isn't required now.